### ISSUE TITLE
Add block group variations to the block toolbar

### ISSUE THREAD

**Issue by richtabor** [WordPress/gutenberg#49524]
## What problem does this address?

There's not a quick way to set blocks as a row, stack, (eventually the grid variation perhaps) when using the "Group" control from a block toolbar's options menu. Currently you have to invoke the Block Inspector to locate the variation icon shortcuts. 

We already have the group variation icons that render with multi-select. I think it'd be nice if those variation icons were also available within the Group block's toolbar — instead of requiring the use of the Block Inspector. The Group block's toolbar only has block alignment, so there's plenty of room. 

## What is your proposed solution?

Add the block's variation icons to the block toolbar, allowing for quicker access to the Row/Stack/Grid icons. It is a bit odd having both the block icon and the "Group" variation, but we could probably use a "reset" type functionality. 

One idea is having "Group" selected initially; I'm not 100% sure about this:

<img width="392" alt="CleanShot 2023-03-31 at 20 59 17" src="https://user-images.githubusercontent.com/1813435/229257900-61651ee7-e5fe-4c3c-9110-1401be545e73.png">

Another is to only show the "Group" control if one of the variations is selected. Don't love how clicking the "Group" icon then removes it:

https://user-images.githubusercontent.com/1813435/229257701-3c89da4a-ef45-40f8-aeb8-fe0d370d2b01.mp4

## References:

1. Grouping blocks via the options menu: 

<img width="619" alt="CleanShot 2023-03-31 at 20 38 09" src="https://user-images.githubusercontent.com/1813435/229256906-95405db3-d418-4d09-aca3-7e221613007a.png">

2. You are then required to use the Block Inspector variation icons, to set a row, stack, or grid variation: 

<img width="1113" alt="CleanShot 2023-03-31 at 20 38 45" src="https://user-images.githubusercontent.com/1813435/229256937-baa263e9-9038-482e-a673-d0f77d6dd788.png">

3. The existing multi-select `BlockGroupToolbar` controls. I'm proposing we have these row, stack, and now grid, variation controls on the core/group block: 

<img width="420" alt="CleanShot 2023-03-31 at 20 11 05" src="https://user-images.githubusercontent.com/1813435/229255149-a605e6f9-0603-4129-bda1-451b74d032be.png">

**Comment by richtabor** [comment]
Thoughts? Have other ideas on how to better surface the block variation controls @WordPress/gutenberg-design?

**Comment by richtabor** [comment]
Here's another idea, though I'm not sure about it (same duplicate group icon issue): 

https://user-images.githubusercontent.com/1813435/229258262-c171d24f-b42a-46a3-98dd-7cf283dc7bb4.mp4

**Comment by paaljoachim** [comment]
Actually I think the last one with the drop down works best. It becomes somewhat similar to the Align icon next to it.
The only thing missing to the align icon is to have a small arrow in the bottom right. Giving the user a hint that there are additional options which one can find by clicking the align or Group variations if it ends up using the same method as align.

Example from Affinity Photo. (Photoshop has similar icons to give a hint that there are additional options available.)
![Screenshot 2023-04-01 at 09 15 06](https://user-images.githubusercontent.com/5323259/229271586-c10cd43d-ac45-4889-87a3-77dd813b173d.jpg)

**Comment by jameskoster** [comment]
I agree the last one is the best of the bunch, but I also wonder whether these should be part of the transform menu? 

<img width="335" alt="Screenshot 2023-04-03 at 13 21 41" src="https://user-images.githubusercontent.com/846565/229507739-c7a5db03-07a6-4d6b-a37d-4c950ed233a1.png">

**Comment by richtabor** [comment]
I they're handy enough to be top-level toolbar controls. 

We really wouldn't need the Group block's placeholder state as-is even, re (https://github.com/WordPress/gutenberg/issues/49525). You would have the toolbar controls in a familiar place, with the existing/familiar iconography.

**Comment by jameskoster** [comment]
> I they're handy enough to be top-level toolbar controls.

I agree, but the transform menu is already a top-level toolbar control? I'm just a little anxious about duplicating an action that already exists in very close proximity. It's not a strong opinion though, the Group toolbar is relatively sparse so the additional control(s) wouldn't add much noise. 

We should probably update the transform options _as well_ :)

> We really wouldn't need the Group block's placeholder state as-is even

It would be nice to get rid of that 'select a layout' step 👍

**Comment by jasmussen** [comment]
Good thoughts all around.

High level we're spanning a bit broadly for the group, with transforms in the inspector, in the ellipsis menu, and in the transforms dialog. There's even an "Unwrap" item now. I think we can consolidate that quite a bit, and have as little as group and ungroup in the ellipsis menu, removing the unwrap (ungroup could unwrap a column if need be). 

I tend to also lean towards the transforms menu being the interface to leverage for this, rather than the block toolbar itself. But that's more of a holistic thought about all transforms, rather than a particular note on this design itself. E.g. if https://github.com/WordPress/gutenberg/issues/46195 becomes a thing more broadly, it's generally good to think of generic interfaces rather than group-specific features.

Finally, the transforms dropdown has been in need of love for a long time. I like to think of it as potentially being the "block control center", so I'd love to see us elevating what's there in that dropdown, and how visual it is. I think it can be an important transformation and previewing interface, holding not just block transforms, but pattern transforms as well. It would need a design pass.

That said, I'm also not strongly _opposed_ to adding these transforms to the toolbar, there's certainly room. I'm just mainly waffling over whether the additional and group-only options are actually helpful or not.

> We really wouldn't need the Group block's placeholder state as-is even, re (https://github.com/WordPress/gutenberg/issues/49525). You would have the toolbar controls in a familiar place, with the existing/familiar iconography.

I actually think it's useful to keep the group block's placeholder state. You'll very rarely see it, due to how grouping can become a thing that happens to other content, or something you get through patterns. But when you _do_ insert a group fresh on its own, it can be a useful _layouting starting point_. As I see it, it doesn't have to be about the group only, it could be a shortcut to multiple layouts, even complex multi-block wireframe patterns.

**Comment by richtabor** [comment]
Ran into this a good bit while building more patterns. It does seem _a bit_ inconsistent that we'd have the transform controls available during multi-select, but not an empty group state. Maybe the existing transform dropdown would be fine, but idk.

<img width="1355" alt="CleanShot 2023-04-19 at 17 48 40" src="https://user-images.githubusercontent.com/1813435/233207731-fa1a5128-ec83-4913-8ffa-04b25d5c757f.png">

**Comment by hanneslsm** [comment]
I also don't feel like the toolbar is the correct place.
Via the toolbar, we can already
- edit "levels" (e.g. H1, H2)
- toggle options (e.g. Image caption, search block)
- change styles (e.g. search block)
- transform the block (https://github.com/WordPress/gutenberg/issues/50025)
- edit the block (e.g. table)
- …

I like that the toolbar is primary for alignments, text styles (bold, italic…), and quick actions (like "Replace" for images)

Also, when building patterns I have the block inspector always open and find transforming groups there quite fast and intuitive:
<img width="278" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/138d90c5-229b-4e83-a722-99d687b23611">
